rackhd_BIST/test_suites.py

```diff
@@ -0,0 +1,608 @@
+            self.unchecked_configs.remove(key)
+            if value == "":  # value is empty means we don't care value of this items
+                pass
+            elif config != value:
+                description = "RackHD configuration item {} is not typical value".format(key)
+                # details = "Default value: {}, user value: {}".format(value, config)
+                details = ''
+                Logger.record_log_message(
+                    description, details, self.log_level[template_key]["unequal"])
+
+    def validate_config_items(self):
+        """
+        Validate RackHD configure files against configure template
+        """
+        for key in self.rackhd_config_template.keys():
+            self.validate_config_via_key(key)
+
+    def logging_extra_items(self):
+        """
+        Record extra items besides template items
+        """
+        for key in self.unchecked_configs:
+            description = "Configuration item {} is specified".format(key)
+            details = "{}: {}".format(key, self.rackhd_config.get(key))
+            Logger.record_log_message(description, details, "info")
+
+    def run_test(self):
+        """
+        Rackhd configure tests
+        """
+        self.load_config_file()
+        self.load_config_template()
+        if not self.rackhd_config_template:
+            return -1
+        if not self.rackhd_config:
+            print "Can't find or load RackHD configuration files, existing tests"
+            sys.exit(-1)  # Without configuration, RackHD services and APIs can't work
+        self.validate_config_items()
+        self.logging_extra_items()
+
+
+class Tools(object):
+    """
+    RackHD required tools test suite
+    """
+    def __init__(self):
+        self.tools = CONFIGURATION.get("toolList")
+
+    def validate_requirement(self, requirement, real_version):
+        """
+        Validate tool version
+        :param requirement: an object includes tool version requirement,
+            it may contains none or any of below items
+            min: required minimum version
+            max: required maximum version
+            among: version should be among a list
+            exclusive: version should not be any of a list
+        :param version: tool version got
+        :return: Boolean value, True for valid version False for invalid version
+        """
+        if not requirement:
+            return True
+        if requirement.has_key("among") and real_version not in requirement["among"]:
+            return False
+        if requirement.has_key("min") and \
+                utils.tool_version_compare(real_version, requirement["min"]) == -1:
+            return False
+        if requirement.has_key("max") and \
+                utils.tool_version_compare(real_version, requirement["max"]) == 1:
+            return False
+        if requirement.has_key("exclusive") and real_version in requirement["exclusive"]:
+            return False
+        return True
+
```

Comment:
What about merging all the if into one with logical condition?
e.g. if XXX and XXX or XXX: return False
else return True